README.md

```diff
@@ -9,3 +9,34 @@ The `materialize-*` connectors all implement the [Flow Materialize gRPC protocol
+  - [ ] If your technology does not support transactions, but does support a retriable idempotent store operation, then you can have the Flow Recovery Log be authoritative and use the idempotency of the operation to ensure exactly-once semantics even in cases of failure. The general idea is to keep track of the operations that are being run in the store phase in the checkpoint sent to the runtime, so that in case the connector fails before it can successfully commit the checkpoint to the recovery log, on the next start of the connector, the operation can be retried by looking at the checkpoint and running the idempotent operations again as part of the Open phase. For an example of this see [materialize-databricks](https://github.com/estuary/connectors/tree/main/materialize-databricks).
+  - [ ] There are certain technologies that will not support either of these, in those cases we cannot guarantee exactly-once semantics, but at-least-once. For an example of this see [materialize-google-pubsub](https://github.com/estuary/connectors/tree/main/materialize-google-pubsub).
+- [ ] Note that in a mater
+- [ ] If your technology supports different modes of authentication, the recommended approach is to have a `oneOf` jsonschema under the key `credentials`. See [materialize-databricks](https://github.com/estuary/connectors/blob/main/materialize-databricks/.snapshots/TestSpecification) as an example.
+- [ ] We recommend using our [schema-gen](https://github.com/estuary/connectors/tree/main/go/schema-gen) if you are using the golang [invopop/jsonschema](https://github.com/invopop/jsonschema) module to generate JSONSchema for your module as it has some extra quality-of-life improvements.
+- [ ] If you want to support network tunnelling access to your technology (at the moment that means connecting through a SSH bastion), you can use the [network-tunnel](https://github.com/estuary/connectors/tree/main/go/network-tunnel) library. See [materialize-postgres](https://github.com/estuary/connectors/tree/main/materialize-postgres) for an example of its usage.
+- [ ] If you want to support Google OAuth and Service Account authentication methods, you can use the [auth/google](https://github.com/estuary/connectors/tree/main/go/auth/google) library.
+
+### SQL Materializations
+
+- [ ] The [materialize-sql](https://github.com/estuary/connectors/tree/main/materialize-sql) library abstracts away a lot of shared logic among our SQL materializations. Connector developers will need to implement the various interfaces of this library, and there are [standard implementations](https://github.com/estuary/connectors/blob/main/materialize-sql/std_sql.go) available as well, but they may or may not work with your destination technology. There are many examples of SQL materializations using this library that you can check to get an idea of how they work.
```

Comment:
I'll add some more to this after I finished my current round of refactoring. There is a common Validation [construct](https://github.com/estuary/connectors/blob/main/materialize-boilerplate/validate/validate.go) that is usable by all materializations (particularly those that produce constraints based on binding changes), and I am working on a somewhat similar mechanism for `Apply`.